// evergreen-ui/commonjs/themes/default/components/badge.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const tinycolor2_1 = __importDefault(require("tinycolor2"));
const theme_tools_1 = require("../../../theme/src/theme-tools");
const baseStyle = {
    height: 16,
    paddingY: 0,
    paddingX: 6,
    borderRadius: 'radii.1',
    fontSize: '11.5px',
    textAlign: 'center',
    textDecoration: 'none',
    textTransform: 'uppercase'
};
const appearances = {
    subtle: (theme, props) => {
        const scheme = (0, theme_tools_1.get)(theme, `fills.${props.color}`, {
            backgroundColor: props.color,
            color: (0, tinycolor2_1.default)(props.color).isLight() ? 'colors.dark' : 'white'
        });
        return {
            color: scheme.color,
            backgroundColor: scheme.backgroundColor
        };
    }
};
const sizes = {};
exports.default = {
    baseStyle,
    appearances,
    sizes
};
